### ISSUE TITLE
Standardize "platform names" for non-x86 architectures

### ISSUE THREAD

**Issue by t-sommer** [modelica/ModelicaSpecification#3453]
Currently the MLS (3.7-dev, p. 205) defines only "platform names" for x86 and x86_64 on Windows and Linux:
 
> – "win32" (Microsoft Windows 32 bit)
> – "win64" (Microsoft Windows 64 bit)
> – "linux32" (Linux Intel 32 bit)
> – "linux64" (Linux Intel 64 bit)

I suggest to adopt the [platform tuple of FMI 3.0](https://fmi-standard.org/docs/3.0.1/#platform-tupe-definition) to support other architectures and operating systems (e.g. macOS on Apple Silicon and Windows on ARM).

**Comment by DagBruck** [comment]
Really, some organization with more authority on the topic than MA should already have decided such a naming convention? Let's just steal something and point to the source.

**Comment by DagBruck** [comment]
> I assume there are similar considerations for gcc.

Well, at least we have the problem with different Linux kernels and libc, although the issues for complete applications with GUI seem more complicated.

**Comment by sjoelund** [comment]
> Well, at least we have the problem with different Linux kernels and libc, although the issues for complete applications with GUI seem more complicated.

Yes, the usual solution is to have a very old Linux system (usually RedHat) compile the library since Linux kernels and libc are mostly backwards compatible. But we would need to use some standardized name. It is crazy that the actual portable way of getting things to work has been the Include annotation since it ignores these issues.

Right now, according to the specification, there is no way to specify Intel or Apple Silicon for Mac (no Mac at all actually).

A standardized BuildProject (#1668) could also resolve things by letting the compiler install the built library anywhere it likes. However, even the MSL CMake attempt fails at this because the Modelica_platform.cmake needs to describe the name of the platform.

https://github.com/modelica/ModelicaStandardLibrary/blob/master/Modelica/Resources/BuildProjects/CMake/Modelica_platform.cmake did choose darwin64 and aarch64-darwin, but that's not standardized. And it assumes any Unix (including BSD) is linux.

Moreover, we have armhf, aarch64 for Linux and then wasm32 as well as wasm32-wasip1 to consider.

So even if the name is bad, choosing the FMI platform name would make sense since it could potentially work. It can't make things any worse.

**Comment by Modelica-ZhipengChen** [comment]
According to https://support.wolfram.com/70669, System Modeler supports macOS on both Intel (x86-64) and Apple Silicon. Could you tell me the platform names used for these two targets? Do they follow the FMI 3.0 platform tuple convention (https://fmi-standard.org/docs/3.0.1/#platform-tuple-examples), i.e., `x86_64-darwin` and `aarch64-darwin`? @henrikt-ma 

In mid-June 2026, MWORKS.Sysplorer completed its adaptation to HarmonyOS and encountered a similar issue. We therefore hope to add an `ohos64_aarch` option. 

Before the `ohos64_aarch` option is available, we have to reuse the linux64 platform tag for HarmonyOS. However, this means the library must be published as two separate packages — one for HarmonyOS and one for Windows/Linux — which is rather inconvenient.

> Related proposal on the FMI side: https://github.com/modelica/fmi-standard/issues/2195

**Comment by maltelenz** [comment]
@Modelica-ZhipengChen System Modeler contributed some [cmake code for ModelicaStandardLibrary](https://github.com/modelica/ModelicaStandardLibrary/blob/master/Modelica/Resources/BuildProjects/CMake/Modelica_platform.cmake#L7-L33), where you can find that we use `darwin64` and `aarch64-darwin`. This might look inconsistent, but we already had `darwin64` long before Apple Silicon was invented.

**Comment by sjoelund** [comment]
Also, darwin64 can work on both x86_64 and aarch64 since macOS often uses universal (fat) binaries with support for both architectures.
OpenModelica has used the same names (also searching both of them). OpenModelica searches `x86_64-linux` in addition to `linux64` also for Modelica.

**Comment by Modelica-ZhipengChen** [comment]
> [@Modelica-ZhipengChen](https://github.com/Modelica-ZhipengChen) System Modeler contributed some [cmake code for ModelicaStandardLibrary](https://github.com/modelica/ModelicaStandardLibrary/blob/master/Modelica/Resources/BuildProjects/CMake/Modelica_platform.cmake#L7-L33), where you can find that we use `darwin64` and `aarch64-darwin`. This might look inconsistent, but we already had `darwin64` long before Apple Silicon was invented.

`darwin64` is indeed a reasonable name — after all, it conforms to the FMI 2.0 specification, and its format is closer to the `win64` and `linux64` conventions in the Modelica specification.

**Comment by Modelica-ZhipengChen** [comment]
> Also, darwin64 can work on both x86_64 and aarch64 since macOS often uses universal (fat) binaries with support for both architectures. OpenModelica has used the same names (also searching both of them). OpenModelica searches `x86_64-linux` in addition to `linux64` also for Modelica.

MWORKS.Sysplorer follows a similar convention on HarmonyOS: it prefers the new name first, and falls back to searching `linux64` if the content is not found.
